apps/src/applab/applab.js

```diff
@@ -815,7 +817,17 @@ Applab.init = function(config) {
   config.varsInGlobals = true;
   config.noButtonsBelowOnMobileShare = true;
 
-  config.dropletConfig = dropletConfig;
+  // filter blocks to exclude anything that isn't in code functions if
+  // autocompletePaletteApisOnly is true
+  dropletConfigBlocks = dropletConfig.blocks.filter(function (block) {
+    return !(config.level.autocompletePaletteApisOnly &&
+      config.level.codeFunctions[block.func] === undefined);
+  });
+
+  config.dropletConfig = $.extend({}, dropletConfig, {
+    blocks: dropletConfigBlocks
+  });
```

Comment:
I considered doing something like that, but that makes our code less modular. One place that could get us into trouble is in mochaTests. Consider this case:
Test 1: Has autocompletePaletteApisOnly set, and mutates dropletConfig to be limited set of blocks
Test 2: Does not have flag set. Calls `require('./dropletConfig')` expecting full set of blocks. Actually gets the filtered set, since when we modified dropletConfig, we were actually modifying the object that will be returned by the require call.